src/EventListener/DeserializeListener.php

```diff
@@ -66,12 +66,31 @@ public function onKernelRequest(GetResponseEvent $event)
         $request->attributes->set(
             'data',
             $this->serializer->deserialize(
-                $request->getContent(), $attributes['resource_class'], $format, $context
+                $this->getRequestContent($request), $attributes['resource_class'], $format, $context
```

Comment:
We can skip the `deserialize`. Actually I'd go as far as to skip `SerializerContextBuilder::createFromRequest` too.

> We lost the loaded entity (we can't retrieve it in "data") in the case of empty put request on specific entity (for exemple PUT /dummies/1/custom_operation)

It's already set in `ReadListener`. Don't overwrite it. :stuck_out_tongue: